lib/ansible/modules/cloud/alicloud/alicloud_instance_facts.py

```diff
@@ -0,0 +1,245 @@
+            "vpc_id": "",
+            "vswitch_id": ""
+        },
+        "zone_id": "cn-beijing-a"
+    }]
+total:
+    description: The number of all instances after operating ecs instance.
+    returned: when success
+    type: int
+    sample: 1
+'''
+
+# import time
+# import sys
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.alicloud_ecs import get_acs_connection_info, ecs_argument_spec, ecs_connect
+
+HAS_FOOTMARK = False
+
+try:
+    from footmark.exception import ECSResponseError
+    HAS_FOOTMARK = True
+except ImportError:
+    HAS_FOOTMARK = False
+
+
+def get_instance_info(inst):
+    """
+    Retrieves instance information from an instance
+    ID and returns it as a dictionary
+    """
+    instance_info = {'id': inst.id,
+                     'private_ip': inst.inner_ip_address,
+                     'public_ip': inst.public_ip_address,
+                     'image_id': inst.image_id,
+                     'zone_id': inst.zone_id,
+                     'region_id': inst.region_id,
+                     'launch_time': inst.creation_time,
+                     'instance_type': inst.instance_type,
+                     'instance_name': inst.instance_name,
+                     'host_name': inst.host_name,
+                     'group_ids': inst.security_group_ids,
+                     'status': inst.status,
+                     'tags': inst.tags,
+                     'vpc_attributes': inst.vpc_attributes,
+                     'eip': inst.eip_address,
+                     'io_optimized': inst.io_optimized,
+                     'key_name': inst.key_name
+                     }
+    try:
+        bdm_dict = {}
+        bdm = getattr(inst, 'block_device_mapping')
+        for device_name in bdm.keys():
+            bdm_dict[device_name] = {
+                'status': bdm[device_name].status,
+                'volume_id': bdm[device_name].volume_id,
+                'delete_on_termination': bdm[device_name].delete_on_termination
+            }
+        instance_info['block_device_mapping'] = bdm_dict
+    except AttributeError:
+        instance_info['block_device_mapping'] = False
+
+    return instance_info
+
+
+def main():
+    argument_spec = ecs_argument_spec()
+    argument_spec.update(dict(
+        alicloud_zone=dict(aliases=['zone_id', 'zone']),
+        instance_ids=dict(type='list', aliases=['ids']),
+        instance_names=dict(type='list', aliases=['names']),
+        instance_tags=dict(type='list', aliases=['tags']),
+    )
+    )
+    module = AnsibleModule(argument_spec=argument_spec)
+
+    if HAS_FOOTMARK is False:
+        module.fail_json(msg='footmark required for the module alicloud_instance_facts')
+
+    ecs = ecs_connect(module)
+
+    instances = []
+    instance_ids = []
+    ids = module.params['instance_ids']
+    names = module.params['instance_names']
+    zone_id = module.params['alicloud_zone']
+    if ids and (not isinstance(ids, list) or len(ids)) < 1:
```

Comment:
This conditional seems incorrect - parenthesis are closed too early. 

I think it should read `if ids and (not isinstance(ids, list) or len(ids) < 1):`